src/features/device-state/routes/fleet-state-get-v3.ts

```diff
@@ -0,0 +1,372 @@
+import type { RequestHandler } from 'express';
+import type { Request } from 'express';
+
+import * as _ from 'lodash';
+import {
+	captureException,
+	handleHttpErrors,
+} from '../../../infra/error-handling';
+import {
+	ConfigurationVarsToLabels,
+	EnvVarList,
+	formatImageLocation,
+	readTransaction,
+	rejectUiConfig,
+	setDefaultConfigVariables,
+	varListInsert,
+} from '../state-get-utils';
+import { Expand } from 'pinejs-client-core';
+
+import { sbvrUtils, errors } from '@balena/pinejs';
+import { Application } from '../../../balena-model';
+const { api } = sbvrUtils;
+const { UnauthorizedError } = errors;
+
+type LocalStateApp = FleetStateV3[string]['apps'][string];
+type ServiceComposition = AnyObject;
+export type FleetStateV3 = {
```

Comment:
Can this be shared with device-state-v3? I think you can do it with a bunch of `{ ...DeviceStateV3['...], x: ... }` until you get to the `Exclude<x, 'is_managed_by__device'>` if you really want, although since that property is optional anyway you shouldn't need to have to remove it?

Reply:
Solved with:
```
type FleetStateV3 = {
	[uuid: string]: Omit<StateV3[string], 'is_managed_by__device' | 'apps'> & {
		apps: {
			[uuid: string]: Omit<
				StateV3[string]['apps'][string],
				'is_managed_by__device'
			>;
		};
	};
};
```